package/MDAnalysis/core/universe.py

```diff
@@ -494,8 +495,22 @@ def transfer_to_memory(self, start=None, stop=None, step=None,
                 coordinates = []  # TODO: use pre-allocated array
                 for i, ts in enumerate(self.trajectory[start:stop:step]):
                     coordinates.append(np.copy(ts.positions))
+                    dimensions.append(np.copy(ts.dimensions))
                     pm.echo(i, frame=ts.frame)
                 coordinates = np.array(coordinates)
+            # If we extrated the coordinates using Timeseries, we still need
+            # to go through the trajectory to extract the dimensions.
+            else:
+                n_frames = len(range(
+                    *self.trajectory.check_slice_indices(start, stop, step)
+                ))
+                pm_format = ('{step}/{numsteps} frame dimensions '
+                             'copied to memory (frame {frame})')
+                pm = ProgressMeter(n_frames, interval=1,
+                                   verbose=verbose, format=pm_format)
+                for i, ts in enumerate(self.trajectory[start:stop:step]):
```

Comment:
btw having the timeseries being able to read fields on demand would allow to add all the same fields to the memory reader, not only the box information